python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -214,6 +214,15 @@ def watch(self, run_id: str, cursor: str, callback: Callable):
     def end_watch(self, run_id: str, handler: Callable):
         """Call this method to stop watching."""
 
+    def supports_watch_asset_events(self):
```

Comment:
do we need this?  It seems like the polling watcher would work for all sql based implementations?

Reply:
Hmm this might be true - I was mostly planning of leaving this out of the LegacyEventLogStorage and InMemoryEventLogStorage, but maybe I can do a basic implementation for those and we'll have complete coverage.

The polling watcher does seem like it would work for all the SQL implementations, but it looks like each concrete class implements these methods and builds an instance of the watcher itself. It looks like the Postgres event log watcher uses a different implementation built on a postgres-specific notification system, so that may be why? I'll check with @gibsondan and see since it looks like he wrote the Postgres one!